src/main/java/org/isf/visits/gui/InsertVisit.java

```diff
@@ -321,44 +307,26 @@ private JButton getButtonOK() {
 			buttonOK.setMnemonic(MessageBundle.getMnemonic("angal.common.ok.btn.key"));
 			buttonOK.addActionListener(new ActionListener() {
 
-				private VisitManager visitManager = Context.getApplicationContext().getBean(VisitManager.class);
-
 				@Override
 				public void actionPerformed(ActionEvent actionEvent) {
-					if (visitDateChooser.getDate() == null) {
-						MessageDialog.error(InsertVisit.this, "angal.visit.pleasechooseadate.msg");
-						return;
-					}
 					GregorianCalendar date = new GregorianCalendar();
 					date.setTime(visitDateChooser.getDate());
-					int visitID = 0;
-					String note = null;
-					Object o = jSpinnerDur.getValue();
-					Number n = (Number) o;
-					int i = n.intValue();
-					String duration = String.valueOf(i);
-					String service = serviceField.getText();
-					Object ward = wardBox.getSelectedItem();
-					if (ward instanceof Ward) {
-						saveWard = getWard();
-					} else {
-						MessageDialog.error(InsertVisit.this, "angal.visit.pleasechooseaward.msg");
-						return;
-					}
-
-					boolean sms = false;
-					if (patientSelected == null) {
-						MessageDialog.error(InsertVisit.this, "angal.visit.pleasechooseapatient.msg");
+					if (date.before(TimeTools.getDateToday0())) {
```

Comment:
The method `actionPerformed` for the `ActionListener` could be a lambda expression.

Reply:
Done, thanks!